### ISSUE TITLE
[MASSEMBLY-754] Non existing folder produces NPE

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-assembly-plugin#966]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** opened **[MASSEMBLY-754](https://issues.apache.org/jira/browse/MASSEMBLY-754?redirect=false)** and commented

If i define a descriptor which uses a not existing folder it will produce a NPE.
Using the following descriptor:

```
<assembly xmlns="http://maven.apache.org/plugins/maven-assembly-plugin/assembly/1.1.0"
xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
xsi:schemaLocation="http://maven.apache.org/plugins/maven-assembly-plugin/assembly/1.1.0 http://maven.apache.org/xsd/assembly-1.1.0.xsd">

  <id>dist-assembly</id>

  <formats>
      <format>tar.gz</format>
  </formats>

  <fileSets>
    <fileSet>
      <directory>folder</directory>
      <outputDirectory>.</outputDirectory>
    </fileSet>
  </fileSets>
</assembly>
```

where the folder `folder` does not exist will produce the failure. If i create the folder with a single file it will not occur.

---

**Affects:** 2.5.3

**Attachments:**
- [massembly-754.tar.gz](https://issues.apache.org/jira/secure/attachment/12714372/massembly-754.tar.gz) (_860 bytes_)

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

Attache full working example of the problem.

**Comment by jira-importer** [comment]
**[Kristian Rosenvold](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=krosenvold)** commented

Npe fixed in ab62f317ffffd8b11017edc2dd7ae46363c09957 in plexus-archiver
